pipert/core/message.py

```diff
@@ -117,13 +118,20 @@ def record_custom(self, component_name, section):
     def record_exit(self, component_name, logger):
         """
         Records the timestamp of the message's exit out of a component.
+        Additionally, it enables a flag called 'reached_exit' if the message is exiting
+        the pipeline's "output component".
 
         Args:
             component_name: the name of the component that the message exited.
             logger: the logger object of the component's output routine.
         """
-        self.history[component_name]["exit"] = time.time()
-        logger.info("Sending the following message: %s", str(self))
+        if "exit" not in self.history[component_name]:
+            self.history[component_name]["exit"] = time.time()
+            if component_name == "FlaskVideoDisplay" or component_name == "VideoWriter":
```

Comment:
There is a problem here because it's very specific and hard coded.
We will need to change this if statement every time to our end component.
It should be generic and get the end components from maybe the Pipe manager..

Reply:
I totally agree - I did it this way simply because currently there is no pipeline manager or a config file that can tell us how the pipeline is configured.

Reply:
Ok.
Open a new issue that depends on the pipeline manager.

Reply:
Done.